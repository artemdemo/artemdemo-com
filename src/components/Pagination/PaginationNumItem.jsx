import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'gatsby';
import * as utils from '../../services/utils';

const PaginationNumItem = (props) => {
    const { pageNum, active } = props;
    const itemClass = classnames('page-item', {
        active
    });
    const path = pageNum > 1 ? utils.prefixBlog(`/${pageNum}`) : '/';

    return (
        <li className={itemClass}>
            <Link
                className='page-link'
                to={path}
            >
                {pageNum}
            </Link>
        </li>
    );
};

PaginationNumItem.propTypes = {
    pageNum: PropTypes.number.isRequired,
    active: PropTypes.bool,
};

PaginationNumItem.defaultProps = {
    active: false,
};

export default PaginationNumItem;