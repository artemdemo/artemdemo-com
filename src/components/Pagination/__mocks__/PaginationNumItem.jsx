import React from 'react';

const PaginationNumItem = props => (
    <div data-mock='PaginationNumItem'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default PaginationNumItem;