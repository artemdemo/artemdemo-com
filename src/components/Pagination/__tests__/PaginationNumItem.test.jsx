import React from 'react';
import renderer from 'react-test-renderer';
import PaginationNumItem from '../PaginationNumItem';

jest.mock('gatsby');

describe('PaginationNumItem', () => {
    it('should render with pageNum', () => {
        const tree = renderer.create(
            <PaginationNumItem
                pageNum={1}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pageNum > 1', () => {
        const tree = renderer.create(
            <PaginationNumItem
                pageNum={2}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pageNum > 1 and active', () => {
        const tree = renderer.create(
            <PaginationNumItem
                pageNum={2}
                active
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with prefix', () => {
        const tree = renderer.create(
            <PaginationNumItem
                pageNum={2}
                prefixer={path => `test-path${path}`}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});