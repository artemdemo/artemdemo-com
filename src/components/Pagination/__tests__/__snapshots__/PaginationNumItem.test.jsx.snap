// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNumItem should render with pageNum > 1 1`] = `
<li
  className="page-item"
>
  <div
    data-mock="Link"
  >
    {
  "className": "page-link",
  "to": "/2"
}
    2
  </div>
</li>
`;

exports[`PaginationNumItem should render with pageNum > 1 and active 1`] = `
<li
  className="page-item active"
>
  <div
    data-mock="Link"
  >
    {
  "className": "page-link",
  "to": "/2"
}
    2
  </div>
</li>
`;

exports[`PaginationNumItem should render with pageNum 1`] = `
<li
  className="page-item"
>
  <div
    data-mock="Link"
  >
    {
  "className": "page-link"
}
    1
  </div>
</li>
`;
